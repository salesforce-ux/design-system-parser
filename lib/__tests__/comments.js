const comments = require('../comments')

it('parses description and annotations', () => {
  expect(comments(`
    /**
     * Hello
     * @world
     */
  `)).toMatchSnapshot()
})

it('parses description and annotations (multi)', () => {
  expect(comments(`
    /**
     * This is a multi-line
     * description
     * @a hello
     * @b world
     */
  `)).toMatchSnapshot()
})

it('ignores annotations inside a code fence', () => {
  expect(comments(`
    /**
     * This comment contains a code fence
     * \`\`\`scss
     * @media () {}
     * \`\`\`
     * @a
     * @b
     */
  `)).toMatchSnapshot()
})
