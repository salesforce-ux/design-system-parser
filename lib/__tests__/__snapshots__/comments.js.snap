// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores annotations inside a code fence 1`] = `
Immutable.List [
  Immutable.Map {
    "description": "This comment contains a code fence
\`\`\`scss
@media () {}
\`\`\`",
    "annotations": Immutable.Map {
      "a": true,
      "b": true,
    },
  },
]
`;

exports[`parses description and annotations (multi) 1`] = `
Immutable.List [
  Immutable.Map {
    "description": "This is a multi-line
description",
    "annotations": Immutable.Map {
      "a": "hello",
      "b": "world",
    },
  },
]
`;

exports[`parses description and annotations 1`] = `
Immutable.List [
  Immutable.Map {
    "description": "Hello",
    "annotations": Immutable.Map {
      "world": true,
    },
  },
]
`;
